File: (7 kyu) Get the Middle Character/(7 kyu) Get the Middle Character.cpp
#include <iostream>
#include <string>

using namespace std;

std::string get_middle(std::string input)
{
    int middle = input.length() / 2;
    string res = "";
    if (input.length() % 2 == 0)
        res += input[middle - 1];
    res += input[middle];
    return res;
}

File: (7 kyu) Get the Middle Character/(7 kyu) Get the Middle Character.cs
public class Kata
{
  public static string GetMiddle(string s)
  {
    int middle = (s.Length - 1)/ 2;
    int len = 1;
    
    if (s.Length % 2 == 0)
      len = 2;
    
    return s.Substring(middle, len);
  }
}
